dbt/include/athena/macros/materializations/snapshots/snapshot.sql

```diff
@@ -1,5 +1,322 @@
+          {% endif %}
+        {% endfor %}
+        FROM snapshotted_data
+        LEFT JOIN source_data
+               ON snapshotted_data.dbt_unique_key = source_data.dbt_unique_key
+        WHERE source_data.dbt_unique_key IS NULL
+    )
+    SELECT * FROM upserts
+    UNION ALL
+    SELECT * FROM deletes;
+    {% else %}
+    SELECT * FROM upserts;
+    {% endif %}
+
+{%- endmacro %}
+
+{%
+  Create a new temporary table to hold the records to be upserted or deleted
+%}
+
+{% macro athena__build_snapshot_staging_table(strategy, sql, target_relation) %}
+    {%- set tmp_identifier = target_relation.identifier ~ '__dbt_tmp' -%}
+
+    {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,
+                                                  schema=target_relation.schema,
+                                                  database=none,
+                                                  type='table') -%}
+
+    {% do adapter.drop_relation(tmp_relation) %}
+
+    {%- set select = snapshot_staging_table(strategy, sql, target_relation) -%}
+
+    {% call statement('build_snapshot_staging_relation') %}
+        {{ create_table_as(False, tmp_relation, select) }}
+    {% endcall %}
+
+    {% do return(tmp_relation) %}
+{% endmacro %}
+
+{#
+    Add new columns to the table if applicable
+#}
+
+{% macro create_columns(relation, columns) -%}
+  {{ adapter.dispatch('create_columns', 'dbt')(relation, columns) }}
+{% endmacro %}
+
+{% macro default__create_columns(relation, columns) -%}
+  {%- for column in columns -%}
+    {% if column.data_type|lower == 'boolean' %}
+    {% set query -%}
+      ALTER TABLE {{ relation }} ADD COLUMNS ({{ column.name }} BOOLEAN);
+    {%- endset -%}
+    {% elif column.data_type|lower == 'character varying(256)' %}
+    {% set query -%}
+      ALTER TABLE {{ relation }} ADD COLUMNS ({{ column.name }} VARCHAR);
+    {%- endset -%}
+    {% elif column.data_type|lower == 'integer' %}
+    {% set query -%}
+      ALTER TABLE {{ relation }} ADD COLUMNS ({{ column.name }} INT);
+    {%- endset -%}
+    {% elif column.data_type|lower == 'float' %}
+    {% set query -%}
+      ALTER TABLE {{ relation }} ADD COLUMNS ({{ column.name }} FLOAT);
+    {%- endset -%}
+    {% else %}
+      ALTER TABLE {{ relation }} ADD COLUMNS ({{ column.name }} {{ column.data_type }});
+    {% endif %}
+    {% do run_query(query) %}
+  {%- endfor %}
+{% endmacro %}
+
+{#
+    Update the dbt_valid_to and is_current_record for
+    snapshot rows being updated and create a new temporary table to hold them
+#}
+
+{% macro athena__create_new_snapshot_table(target, source) %}
+    {%- set tmp_identifier = target.identifier ~ '__dbt_tmp_1' -%}
+
+    {%- set tmp_relation = adapter.get_relation(database=none, schema=target.schema, identifier=tmp_identifier) -%}
+
+    {%- set target_relation = api.Relation.create(identifier=tmp_identifier,
+      schema=target.schema,
```

Comment:
could you refactor this to make use of the default macro `generate_schema_name`??
e.g. quite some of us have custom schema macros to leverage development schemas. I tried with my setup and the macro is not called at all.

Reply:
I add a look, this seems missing totally in dbt-core, please ignore it.